content/news/2023-03-04-bevy-0.10/index.md

```diff
@@ -23,6 +23,42 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
 
 Description here.
 
+## ECS Optimizations
+
+<div class="release-feature-authors">authors: @james7132, @JoJoJet</div>
+
+ECS underlies the entire engine, so eliminating overhead in the ECS results in engine-wide speedups. In 0.10, we've found quite a few areas where we were able to massively reduce the overhead and improve CPU utilization for the entire engine.
+
+In [#6547](https://github.com/bevyengine/bevy/pull/6547), we enabled [autovectorization](https://en.wikipedia.org/wiki/Automatic_vectorization) when using `Query::for_each`, and its parallel variants. Depending on the target architecture the engine is being compiled for, this can result in a 50-87.5% speed up in query iteration time. In 0.11, we may be extending this optimization to all iterator combinators based on `Iterator::fold`, like `Iterator::count`. See [this PR](https://github.com/bevyengine/bevy/pull/6773) for more details.
+
+In [#6681](https://github.com/bevyengine/bevy/pull/6681), by tightly packing entity location metadata and avoiding extra memory lookups, we've significantly reduced the overhead when making random query lookups via `Query::get`, seeing up to a 43% reduction in the overhead spent in `Query::get` and `World::get`.
+
+In [#6800](https://github.com/bevyengine/bevy/pull/6800) and [#6902](https://github.com/bevyengine/bevy/pull/6902), we've found that rustc can optimize out compile-time constant branches across function boundaries, moving the branch from runtime to compile time, has resulted in up to a 50% reduction in overhead when using `EntityRef::get`, `EntityMut::insert`, `EntityMut::remove`, and their variants.
+
+In [#6391](https://github.com/bevyengine/bevy/pull/6391), we've reworked `CommandQueue`'s internals to be more CPU-cache friendly, which has shown up to a 37% speedup when encoding and applying commands.
+
+## Adaptive Batching for Parallel Query Iteration
+
+<div class="release-feature-authors">authors: @james7132</div>
+
+In 0.9 and before, `Query::par_for_each` required providing a batch size, which would be used to chunk the matched entities into batches for parallel processing. This often resulted in users just randomly picking a value and rolling with it, or fine tuning the value based on their development machines. Unfortunately, the most effective value is dependent on the runtime environment (i.e. how many logical cores does a player's computer have) and the state of the ECS World (i.e. how many entities are matched?). Ultimately most users of the API just chose a flat number and lived with the results, good or bad.
```

Comment:
Hope you don't mind a bit of constructive criticism.
I think the flow of the first sentence here could be improved a bit. If I were writing this, I would start out by briefly describing how parallel batches work (hook the reader with exciting performance details), then mention that we used to require an extra parameter for this. That would flow a lot better into the second sentence.

Reply:
I like that a lot! I think the use of language is a lot more engaging now. One thing: you wrote "when their ECS world gets too big to run single threaded". I think it would be more accurate to say "when their queries get too big to run single-threaded"